packages/webapp/src/pages/election/index.tsx

```diff
@@ -20,38 +20,36 @@ export const ElectionPage = () => {
             }
         >
             {isLoading ? (
-                <Container>
-                    <Loader />
-                </Container>
+                <LoaderSection />
             ) : (
                 <div className="divide-y">
                     <Container>
                         <Heading size={1}>Election</Heading>
                     </Container>
-                    <ElectionBody
-                        electionState={currentElection?.electionState}
-                    />
+                    <ElectionBody election={currentElection} />
                 </div>
             )}
         </FluidLayout>
     );
 };
 
-const ElectionBody = ({ electionState }: { electionState?: string }) => {
-    switch (electionState) {
+export default ElectionPage;
+
+const ElectionBody = ({ election }: { election: any }) => {
+    switch (election.electionState) {
         case ElectionStatus.Registration:
         case ElectionStatus.Seeding:
             return <RegistrationElection />;
         case ElectionStatus.Active:
         case ElectionStatus.Final: // TODO: This is one state where there's a board but no satoshi. UI should reflect that.
-            return <OngoingElection />;
+            return <OngoingElection election={election} />;
         default:
-            return (
-                <Container>
-                    <Heading size={2}>Processing election</Heading>
-                </Container>
-            );
+            return <LoaderSection />;
```

Comment:
do we have a way to get no election for some query error and keep showing the loader forever? should we account errors? (not very critical I think)

Reply:
Yeah, I added fetch error handling elsewhere, but not here at this top level. Good catch. In this case, I think what would happen is that the loading spinner _and_ the contents would just go away and the user would be left wondering what happened. I'll add a small error message here.